### ISSUE TITLE
SystemC 2.3.3 - Fatal: (F4) assertion failed: iface != 0

### ISSUE THREAD

**Issue by sedlakmilan** [accellera-official/systemc#81]
Dear all,

Lately, we have been facing a strange issue, causing a "Fatal: (F4) assertion failed: iface != 0" when running our program on a Windows machine.
After removing all functionality and having only the minimum code, we have been able to find out the issue.
The code below is possible to build without any problems.

<img width="1155" alt="Screenshot 2024-04-26 at 12 02 16" src="https://github.com/accellera-official/systemc/assets/127394220/b224236f-31ad-44ea-8442-0e3c0061499b">

But once you run it, you get:
<img width="618" alt="Screenshot 2024-04-26 at 12 04 37" src="https://github.com/accellera-official/systemc/assets/127394220/a5e47bae-e190-4767-826f-fcbaa2f43a43">

<img width="392" alt="Screenshot 2024-04-26 at 12 04 55" src="https://github.com/accellera-official/systemc/assets/127394220/84943437-507a-4547-929e-a72999caa1cb">

Interestingly, if you run this code on a Linux machine using GCC, the code can run without any troubles...

....
After removing the "some_var" in the Utest class, the code is running... 
- It doesn't matter if the "some_var" is private, protected, public... if it's used, initialized in the constructor, etc.
- Once we create a simple variable (any type), inside of the Utest class, the code fails to run (Fatal: (F4) ... )
- If you create a function in the Utest class, the code also runs...

To put it simply, creating a single member variable inside of a class has the result of a "Fatal: (F4) assertion failed: iface != 0", but only on Windows (not Linux with GCC)..

Now, we can fix our full code just by removing the variable, but we just don't understand what the problem is...
Is it a bug? Do we do something that is not allowed?...
Maybe you have some more details about this?

Thank You.
Kind Regards,
Milan

**Comment by AmeyaVS** [comment]
I can confirm this observation on Windows with MSVC.
It probably is hitting some memory alignment issues or probably the project configuration doesn't match the SystemC library build.
Someone who uses day-to-day SystemC on MSVC on Windows would be able to provide a much better explanation.

**Comment by maehne** [comment]
Could you please try whether this issue persists in the latest SystemC PoC library release [3.0.0](https://github.com/accellera-official/systemc/releases/tag/3.0.0)?

**Comment by AmeyaVS** [comment]
@maehne , this is observable even with the latest SystemC commit on the main branch: bec1010.
on Windows with latest MSVC installation.

**Comment by lmailletcontoz** [comment]
Thanks for the feedback. Could you please share the command lines log for the build, to facilitate analysis?

**Comment by lmailletcontoz** [comment]
> Are you looking out for exact compilation commands used to build the project sources?
yes, please. The full log with the options would be appreciated.
